src/controller/CommissionerDiscoveryController.cpp

```diff
@@ -112,6 +112,8 @@ void CommissionerDiscoveryController::OnUserDirectedCommissioningRequest(UDCClie
 
 void CommissionerDiscoveryController::Ok()
 {
+    chip::DeviceLayer::StackLock lock;
```

Comment:
This is assuming it's running on a setup where StackLock is not a no-op, no?  That's not true for controller platforms in general.

The right way to run things that need to be run in the Matter context is via async dispatch to the Matter event loop, if you want it to work across all platforms.

Reply:
I updated these commands to spawn a task on the main chip thread.